core/src/test/java/com/linecorp/armeria/client/AdditionalAuthorityTest.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Test
+    void noAuthority() {
+        final HttpRequest request = HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "/"));
+        assertThatThrownBy(() -> client.execute(request).aggregate().join())
+                .isInstanceOf(CompletionException.class)
+                .hasCauseInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Scheme and authority must be specified in");
+    }
+
+    @Test
+    void absolutePath() {
+        final HttpRequest request =
+                HttpRequest.of(RequestHeaders.of(HttpMethod.GET, "http://foo:" + serverAPort));
+        assertThat(client.execute(request).aggregate().join().contentUtf8()).isEqualTo(
+                "foo/foo:" + serverAPort);
+    }
+
+    @Test
+    void shouldIgnoreInvalidAdditionalAuthority() {
+        // Missing a closing bracket
+        try (SafeCloseable ignored = Clients.withContextCustomizer(
+                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY, "[::1"))) {
+
+            // An invalid authority should be ignored.
+            assertThat(client.get("http://foo:" + serverAPort).aggregate().join().contentUtf8())
+                    .isEqualTo("foo/foo:" + serverAPort);
+        }
+
+        // Port only
+        try (SafeCloseable ignored = Clients.withContextCustomizer(
+                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY, ":8080"))) {
+
+            final HttpRequest request = HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
+                                                                     .scheme("http")
+                                                                     .authority("bar:" + serverAPort)
+                                                                     .build());
+            // If additionalRequestHeader's authority is invalid but req.authority() is valid
+            assertThat(client.execute(request).aggregate().join().contentUtf8())
+                    .isEqualTo("bar/bar:" + serverAPort);
+        }
+
+        // Missing a port number
+        try (SafeCloseable ignored = Clients.withContextCustomizer(
+                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY, "foo:"))) {
+
+            final HttpRequest request = HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
+                                                                     .scheme("http")
+                                                                     .authority("bar:" + serverAPort)
+                                                                     .build());
+            // If additionalRequestHeader's authority is invalid but req.authority() is valid
+            assertThat(client.execute(request).aggregate().join().contentUtf8())
+                    .isEqualTo("bar/bar:" + serverAPort);
+        }
+    }
+
+    @Test
+    void shouldNotUseAuthorityAsEndpointWithBaseUriWebClient() {
```

Comment:
Question: Why do we want to prohibit a user from overriding the authority in this case?

Reply:
If a user proxies requests to another endpoint, the incoming authority could override the target endpoint and the requests could be inifinity loops.
See https://github.com/line/armeria/pull/2757 for details.